src/standalone/Soil/energy_hydrology.jl

```diff
@@ -834,6 +833,18 @@ function turbulent_fluxes!(
     # Obtain surface quantities needed for computation; these should not allocate
     T_sfc = ClimaLand.surface_temperature(model, Y, p, t)
     h_sfc = ClimaLand.surface_height(model, Y, p)
+    ρ_sfc = ClimaLand.surface_air_density(atmos, model, Y, p, t, T_sfc)
+
+    # The following updates q_sfc in place, to reduce allocations
+    # but the original design was not an "update in place" function,
+    # it instead allocated and returned the field.
+    # This needs to be updated everywhere, and then these functions
+    # can become e.g. surface_specific_humidity!(p.soil.q_sfc, ...)
+    ClimaLand.surface_specific_humidity(model, Y, p, T_sfc, ρ_sfc)
```

Comment:
This also updates p.soil.ice_frac in place, and that seems confusing. I will move this into a different function and make it obvious (but am going to let the current long run keep going)